diff --git a/node_modules/zyre.js/lib/zbeacon.js b/node_modules/zyre.js/lib/zbeacon.js
index 0f40c56..f6263d3 100644
--- a/node_modules/zyre.js/lib/zbeacon.js
+++ b/node_modules/zyre.js/lib/zbeacon.js
@@ -6,7 +6,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-const dgram = require('dgram');
+const dgram = require('react-native-udp');
 const ZHelper = require('./zhelper');
 
 const debug = ZHelper.debug('zyre:zbeacon');
diff --git a/node_modules/zyre.js/lib/zhelper.js b/node_modules/zyre.js/lib/zhelper.js
index dca9e9a..0f8ce1a 100644
--- a/node_modules/zyre.js/lib/zhelper.js
+++ b/node_modules/zyre.js/lib/zhelper.js
@@ -6,8 +6,8 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
-const os = require('os');
-const net = require('net');
+const os = require('react-native-os');
+const net = require('react-native-tcp-socket').default;
 const debug = require('debug');
 
 /**
diff --git a/node_modules/zyre.js/lib/zre_msg.js b/node_modules/zyre.js/lib/zre_msg.js
index 86cb9f0..f4a953a 100644
--- a/node_modules/zyre.js/lib/zre_msg.js
+++ b/node_modules/zyre.js/lib/zre_msg.js
@@ -162,7 +162,7 @@ class ZreMsg {
    */
   async send(socket) {
     const msg = this.content ? [this.toBuffer(), this.content] : this.toBuffer();
-    await socket.send(msg);
+    await socket.sendBuffer(msg);
     return this.cmd;
   }
 
diff --git a/node_modules/zyre.js/lib/zyre.js b/node_modules/zyre.js/lib/zyre.js
index 1f7aaa4..92aecee 100644
--- a/node_modules/zyre.js/lib/zyre.js
+++ b/node_modules/zyre.js/lib/zyre.js
@@ -7,6 +7,7 @@
  */
 
 const EventEmitter = require('events');
+require('react-native-get-random-values');
 const { v4: uuidv4 } = require('uuid');
 const ZHelper = require('./zhelper');
 const ZyrePeers = require('./zyre_peers');
diff --git a/node_modules/zyre.js/lib/zyre_node.js b/node_modules/zyre.js/lib/zyre_node.js
index d5b3ccb..e46ecef 100644
--- a/node_modules/zyre.js/lib/zyre_node.js
+++ b/node_modules/zyre.js/lib/zyre_node.js
@@ -7,7 +7,7 @@
  */
 
 const EventEmitter = require('events');
-const zeromq = require('zeromq');
+const zeromq = require('@olehs/react-native-zeromq').ZeroMQ;
 const ZreMsg = require('./zre_msg');
 const ZHelper = require('./zhelper');
 
@@ -65,7 +65,7 @@ class ZyreNode extends EventEmitter {
   async startListening() {
     this._zyrePeers.on('new', this._newPeerHandler);
 
-    this._socket = new zeromq.Router({ handover: true });
+    this._socket = await zeromq.Router({ handover: true });
 
     const receive = async () => {
       try {
diff --git a/node_modules/zyre.js/lib/zyre_peer.js b/node_modules/zyre.js/lib/zyre_peer.js
index 4aeca89..bab407c 100644
--- a/node_modules/zyre.js/lib/zyre_peer.js
+++ b/node_modules/zyre.js/lib/zyre_peer.js
@@ -7,7 +7,7 @@
  */
 
 const EventEmitter = require('events');
-const zeromq = require('zeromq');
+const zeromq = require('@olehs/react-native-zeromq').ZeroMQ;
 const ZreMsg = require('./zre_msg');
 const ZHelper = require('./zhelper');
 
@@ -78,8 +78,8 @@ class ZyrePeer extends EventEmitter {
    */
   async connect() {
     if (typeof this._socket === 'undefined') {
-      this._socket = new zeromq.Dealer({
-        routingId: Buffer.concat([Buffer.from([ID_PREFIX]), this._originID]),
+      this._socket = await zeromq.Dealer({
+        identity: Buffer.concat([Buffer.from([ID_PREFIX]), this._originID]),
         linger: 0,
       });
 
